### DISCUSSION TITLE
UEFA EURO 2024

### DISCUSSION THREAD

**Discussion by msiggi** [OpenLigaDB/OpenLigaDB-Samples#79] [Announcements]
Die "offizielle" Liga zur Europameisterschaft 2024 ist die mit dem Shortcut "em" und der Saison "2024".
Also können alle beim Eingeben der Ergebnisse unter https://www.openligadb.de/Matches/em/2024 helfen. Der Abruf der Daten per Api funktioniert dann z.B. per [https://api.openligadb.de/getmatchdata/em/2024/1](https://api.openligadb.de/getmatchdata/em/2024/1)

**Comment by greynor** [comment]
Ich versuch öfter die Bearbeitungsseite der Ergebnisse zu öffnen, aber er lädt sich einen Wolf. Leider nach fast 20 Minuten ist HUN - SUI immer noch nicht pflegbar

**Reply by msiggi** [reply]
ja, es gab einen Fehler, es sollte nun behoben sein!

**Reply by greynor** [reply]
Besten Dank! :)

**Comment by xdream77** [comment]
Auch heute bei Ukraine Rumänien lädt die Seite nicht durch. 

Meine Google Chrome Konsole spuckt folgendes aus: 
_Third-party cookie will be blocked in future Chrome versions as part of Privacy Sandbox._

Ich habe keine Adblocker, bin nicht im privacy mode und blocke selber nichts. 

Was mir allerdings aufgefallen ist, dass die Seite zwar über **https** geladen wird, manche Landesflaggen aber über **http**. Das kann zu Problemen führen.

**Comment by aha5811** [comment]
Hallo, bei mir lädt die Seite zwar, ich verstehe die Ergebnisse, die man via https://api.openligadb.de/getmatchdata/em/2024/1 erhält, aber nicht. Wieso gibt es goals bei denen matchMinute null eingetragen ist? Warum gibt es vor Ende der 1. Halbzeit in den results schon das result "Halbzeitergebnis" bzw. vor Abpfiff schon das result "Endergebnis"? Liegt das daran, dass die Daten über die dafür nicht geeignete Oberfläche https://www.openligadb.de/Matches/em/2024/1 eingegeben werden? Irgendjemand ("devgass", "Giz") hat anscheinend die Daten dann korrigiert. Gibt es dafür eine andere Oberfläche? Kann ich das als registrierter Benutzer auch? Btw: Im Spiel Rumänien : Ukraine wurde ein Ergebnis "nach Elfmeterscheißen" eingetragen - zum einen der Rechtschreibfehler ;-) zum anderen gibts in der Vorrunde (eigentlich) kein Elfmeterschießen.

**Reply by xdream77** [reply]
Ich bin nicht der Betreiber des Programms, aber ein begeisterter Benutzer.
Ich versuche es mal zu Erklären. 

Bei OpenLigaDB kann jeder angemeldete Nutzer jeden Datensatz (bis zu einem gewissen Grad) verändern.
Jeder kann es kostenlos abrufen und damit seine Seite betreiben.
Das ist von Nutzern für Nutzer.

Wenn du denkst, dieser eine Datensatz hat nicht die erforderlichen Infos, die du gerade brauchst,
dann füge sie hinzu. 
Wenn du denkst, der eine Datensatz ist falsch, dann korrigiere ihn.
Du machst es dadurch gleichzeitig besser für alle anderen.

Mitmachen ist der Schlüssel hier.
LG

**Reply by aha5811** [reply]
Du hast meinen Kommentar entweder nicht gelesen oder ich hab mich missverständlich ausgedrückt.
1. Die Daten machen keinen Sinn:
Sobald man ein Tor einträgt wird der Spielstand danach auch als Halbzeit- und Endergebnis eingetragen. Damit weiß man z.B. wenn das Spiel in der Verlängerung ist und 1:1 eingetragen ist und bei beiden Toren keine matchMinute eingetragen wurde, dann ist matchFinished false und man weiß gar nicht, ob das Spiel schon in der Verlängerung ist oder nicht.
Meiner Meinung nach sollte das Halbzeit-result erst in den results existieren, wenn die Halbzeit abgepfiffen ist, das Endergebnis erst wenn die 2. Halbzeit abgepfiffen ist, vorher muss man sich das Ergebnis eben aus den goals holen.
Jetzt auch wieder Österreich:Frankreich: Da steht Halbzeitstand 0:1, Endstand 0:1 und das Tor dazu (das neben Endstand 0:1 steht), ist in der 38. Minute eingetragen - das müsste neben Halbzeitstand 0:1 stehen.
2. Anscheinend kann eben *nicht* jeder die Daten eintragen/ändern. Man kann zwar immer Tore hinzufügen und auch eintragen, dass das z.B. der Stand nach Elfmeterschießen ist - ich habe aber zumindest nicht herausgefunden, wie man einen falschen Eintrag rückgängig machen kann, also z.B. ein Tor wieder löschen kann oder das Ergebnis nach Elfmeterschießen von Rumänien:Ukraine - vorhin wurde bei dem Spiel mit Belgien in 1:1 eingetragen und ich habe versucht, das zu löschen, das ging aber nicht für mich, und dann hats jemand anders gelöscht.
3. Ich kann nicht ändern, dass da Elfmeterscheißen statt Elfmeterschießen steht.

**Comment by Kristall-Rainer** [comment]
Hallo Alexander,

es hat sich seit Beginn etabliert, dass das Endergebnis das aktuelle Ergebnis ist. Wenn ein Tor in der 1. Hälfte fällt, dann wird das auch als Halbzeitergebnis eingetragen. Somit muss man sich eben nicht alle goals holen (Manche brauchen die nämlich nicht, sondern nur die Ergebnisse). Das Endergebnis ist also immer das aktuelle Ergebnis (quasi Endergebnis Stand jetzt - genauso bei Halbzeitergebnis). Wenn ein Tor in der Verlängerung fällt, wird dieses Ergebnis sowohl als Endergebnis markiert, also auch als „nach Verlängerung“. Der Vorteil dieser Methode ist, dass du ganz genau weißt, in welcher der Halbzeiten (1., 2. oder Verlängerung oder Elfmeterschießen) das Tor gefallen ist, auch wenn die Minute nicht eingetragen ist (das machen in der Tat nicht alle Benutzer) und wie das aktuelle Ergebnis ist.

Viele Grüße,

Matthias

**Reply by aha5811** [reply]
OK, ist zwar komisch, aber wenn das historisch ist, dann ist das so. Man will ja auch nicht allen Code, der darauf vertraut, kaputt machen. 

Die Inkonsistenzen bei den Eintragenden machen mich halt fertig: 
Italien : Albanien 15.06.24 21:00: Endergebnis 2:1 Nicolo Barella 16
vs
Spanien : Kroatien 15.06.24 18:00: Halbzeitergebnis 3:0 Carvajal 47 Overtime, Endergebnis 3:0
vs
Deutschland : Schottland 14.06.24 21:00 Halbzeitergebnis 3:0 Kai Havertz 45 Overtime

Da wird einmal das Tor dem Endergebnis zugeordnet, obwohl es vor der Halbzeit gefallen ist, und einmal dem Halbzeitergebnis; bei dem einen werden die Spielminuten wirklich eingetragen (47 Overtime), einmal nicht (45 Overtime - das Tor ist aber nicht in der 45. gefallen).
Das kann man sich zwar alles irgendwie trotzdem zusammenbasteln, aber es wird halt knifflig.
Mein Anwendungsfall ist: Für ein Tippspiel will ich die Live-Ergebnisse. Bei diesem Tippspiel ist der Stand vor Verlängerung Tipp-relevant, alles was danach passiert ist für die Bewertungen egal, ist aber für die Platzierung der Mannschaften in den einzelnen Gruppen natürlich wichtig. Ich könnte natürlich jetzt einfach abwarten, bis matchFinished und mir dann das Endergebnis holen, es ist aber lustiger, wenn man während des Spiels schon die Bewertungen basierend auf dem aktuellen Spielstand sieht - und die dürfen sich halt in der Finalrunde nach 90min nicht mehr ändern. Mein aktueller Code dazu ist:
1. Sortiere alle Tore nach Spielminute, falls die bei 2 Toren gleich ist (45+Overtime, 90+Overtime), dann sortiere nach Torsumme. Falls bei einem Tor Minute > 90 ohne Overtime ist, dann sind wir in der Verlängerung, der Stand vorher ist das Tipp-relevante Endergebnis und kann fest eingetragen werden. Die Methode kann schiefgehen, wenn ein Tor ohne Minuten eingetragen ist.
2. Wenn das Spiel beendet ist, dann hole das Tipp-relevante Ergebnis aus dem Endergebnis. Das für das Fortkommen der Mannschaften wichtige Ergebnis hole aus dem Tor mit der höchsten Torsumme (alternativ aus dem result mit der höchsten Torsumme).

(Immerhin heißt es jetzt nicht mehr Elfmeterscheißen ;-))

**Comment by Kristall-Rainer** [comment]
Das einzige falsche an den Beispielen unten ist, dass für Havertz Tor nicht die korrekte Minute eingetragen wurde (45. Minute mit dem Hinweis Overtime macht keinen Sinn).
Ansonsten wird das aktuelle Ergebnis immer als Endergebnis markiert. Wenn das Tor in der 1. Halbzeit geschossen wird, dann wird das aktuelle Ergebnis zusätzlich als Halbzeitergebnis markiert. Je nachdem, ob man das Tor zuerst als Halbzeit- oder Endergebnis markiert, steht der Torschütze und die Minute bei openligadb bei Endergebnis oder Halbzeitergebnis. Das ist aber für die API völlig unerheblich, weil dort die Tore und die Ergebnisse unabhängig geliefert werden.


Für deinen Anwendungsfall würde ich einfach resultTypeID=2 nehmen. Das ist das aktuelle Ergebnis vor der Verlängerung. 

Um zu schauen, welche Mannschaft weitergekommen ist, holst du dir, wenn matchFinished=1 noch zusätzlich resultTypeID=2 und resultTypeID=3 und rechnest für jeden Type pointsTeam1 + pointsTeam2. Das höchste Ergebnis ist entscheidend und wenn pointsTeam1>pointsTeam2, dann kommt Team1 weiter, ansonsten Team2.

**Comment by mia66ko** [comment]
Wie kann man den fehlenden Torschützen eintragen?

![TorschuetzeFehlt](https://github.com/OpenLigaDB/OpenLigaDB-Samples/assets/173205552/7239ccf1-e952-4f68-b6f5-c599352e408a)

**Reply by msiggi** [reply]
Per Klick auf die Zeile des Tores, es öffnet sich dann folgender Dialog:
![image](https://github.com/OpenLigaDB/OpenLigaDB-Samples/assets/7813965/7fcf8ae0-b26f-4e6d-a5ad-59b443f32245)
